components/blog/SubscribeBtn.tsx

```diff
@@ -1,14 +1,160 @@
 'use client';
 
-import React from 'react';
+import React, { useRef, useState } from 'react';
 import styled from 'styled-components';
 
-const SubscribeBtn = () => {
-  return <SubscribeBtnContainer>팀 소식 받아보기</SubscribeBtnContainer>;
+import { postSubscriber } from '@/api/blogHome';
+import { IcClose32Icon } from '@/public/icons';
+
+interface subscribeBtnProps {
+  team: string;
+}
+
+const SubscribeBtn = (subscribeBtnProps: subscribeBtnProps) => {
+  const { team } = subscribeBtnProps;
+
+  const [email, setEmail] = useState('');
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+
+  const modalRef = useRef<HTMLDivElement>(null);
+
+  const requestBody = {
+    email: email,
+    blogUrl: team,
+  };
+
+  const handleSubscribe = () => {
+    postSubscriber(requestBody);
+    setModalIsOpen(false);
+  };
+
+  const handleModalCloseBtn = () => {
+    setModalIsOpen(false);
+  };
+
+  const handleModalClose = (e: React.MouseEvent<HTMLElement>) => {
+    !modalRef.current?.contains(e.target as Node) && setModalIsOpen(false);
+  };
+
+  const saveEmail = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(e.target.value);
+  };
+
+  const resetEmail = () => {
+    setEmail('');
+  };
+
+  return (
+    <>
+      {modalIsOpen ? (
+        <ModalBackground onClick={handleModalClose}>
+          <EmailModal ref={modalRef}>
+            <ModalTitleWrapper>
+              <ModalTitle>{team} 아티클 구독하기</ModalTitle>
+              <ModalCloseBtn onClick={handleModalCloseBtn} />
+            </ModalTitleWrapper>
+            <ModalEmailTitle>이메일</ModalEmailTitle>
+            <ModalEmailInput type="text" value={email} onChange={saveEmail}></ModalEmailInput>
+            <ModalBtnWrapper>
+              <ModalSubmitBtn onClick={handleSubscribe}>구독</ModalSubmitBtn>
+            </ModalBtnWrapper>
+          </EmailModal>
+        </ModalBackground>
+      ) : (
+        <></>
+      )}
```

Comment:
```suggestion
      {modalIsOpen && (
        <ModalBackground onClick={handleModalClose}>
          <EmailModal ref={modalRef}>
            <ModalTitleWrapper>
              <ModalTitle>{team} 아티클 구독하기</ModalTitle>
              <ModalCloseBtn onClick={handleModalCloseBtn} />
            </ModalTitleWrapper>
            <ModalEmailTitle>이메일</ModalEmailTitle>
            <ModalEmailInput type="text" value={email} onChange={saveEmail}></ModalEmailInput>
            <ModalBtnWrapper>
              <ModalSubmitBtn onClick={handleSubscribe}>구독</ModalSubmitBtn>
            </ModalBtnWrapper>
          </EmailModal>
        </ModalBackground>
      )}
```

요래요래 바꿔주어도 좋을 것 같습니다!